src/compas_fab/backends/ros/backend_features/move_it_iter_inverse_kinematics.py

```diff
@@ -0,0 +1,91 @@
+              Defaults to the model's root link.
+            - ``"avoid_collisions"``: (:obj:`bool`, optional) Whether or not to avoid collisions.
+              Defaults to ``True``.
+            - ``"constraints"``: (:obj:`list` of :class:`compas_fab.robots.Constraint`, optional)
+              A set of constraints that the request must obey.
+              Defaults to ``None``.
+            - ``"attempts"``: (:obj:`int`, optional) The maximum number of inverse kinematic attempts.
+              Defaults to ``8``.
+            - ``"attached_collision_meshes"``: (:obj:`list` of :class:`compas_fab.robots.AttachedCollisionMesh`, optional)
+              Defaults to ``None``.
+            - ``"max_reach"``: (:obj:`float`, optional): TODO
+            - ``"link_name:``: (:obj:`str`, optional): TODO
+
+        Raises
+        ------
+        compas_fab.backends.exceptions.BackendError
+            If no configuration can be found.
+
+        Yields
+        -------
+        :obj:`tuple` of :obj:`list`
+            A tuple of 2 elements containing a list of joint positions and a list of matching joint names.
+        """
+        # TODO an example
+        options = options or {}
+        max_reach = options.get('max_reach')  # TODO default value?
+        link_name = options.get('link_name')  # TODO or default value here?
+        constraints = options.setdefault('constraints', [])
+
+        group = group or robot.main_group_name
+
+        if max_reach and link_name:
+            # forcing diverse IK solutions by constraining the link_name within a sphere
+            #
+            # here I just want to bring up a discussion point:  since inverse_kinematics is called, and one might
+            # expect there to be constraints already in options, does it really make sense to add these two new
+            # parameters of link_name and max_reach?  maybe it would be more general if the user would feel free
+            # to generate his own constraints, and there could be a flag in the options on whether to generate
+            # random start configurations.
+            assert(link_name in robot.get_link_names(group=group))
+            sphere = Sphere(frame_WCF.point, max_reach)
+            constraints.append(PositionConstraint.from_sphere(link_name, sphere))
+        else:
+            # forcing diverse IK solutions by randomizing the start configuration
+            start_configuration = robot.random_configuration(group=group)
+
+        while True:
+            try:
+                yield self.ros_client.inverse_kinematics(robot,
+                                                         frame_WCF,
+                                                         start_configuration=start_configuration,
+                                                         group=group,
+                                                         options=options)
+            except BackendError:
+                yield "hello"  # TODO what should happen here?
```

Comment:
I have no clue about this. Can it be the same as in the "normal" IK function?
```
compas_fab.backends.exceptions.BackendError
If no configuration can be found.
```

Reply:
If you don't catch the error, then it would make writing code like this next to useless:
```
for configuration in client.iter_inverse_kinematics(robot, frame, ...):
    # do something
```

Reply:
you're right